server/src/test/java/org/elasticsearch/transport/SniffConnectionStrategyTests.java

```diff
@@ -467,7 +470,16 @@ public void testConnectFailsWithIncompatibleNodes() {
                     PlainActionFuture<Void> connectFuture = PlainActionFuture.newFuture();
                     strategy.connect(connectFuture);
 
-                    expectThrows(Exception.class, connectFuture::actionGet);
+                    final NoSeedNodeLeftException exception = (NoSeedNodeLeftException) expectThrows(
+                        ExecutionException.class,
+                        connectFuture::get
+                    ).getCause();
```

Comment:
Sniff mode was throwing a `ConnectTransportException` exception directly. As commented above, I changed it to be wrapped inside a `NoSeedNodeLeftException`.